### PR TITLE
rgw: rgw_auth.cc: disable false use-after-move clang-tidy warning

### PR DESCRIPTION
**Pull request by Suyashd999**
Clang-tidy warning re strategy_result is incorrect. No path leads to a moved-from object
being reused. strategy_result is immediately being re-assigned after the move.

Added annotations to prevent the false detection.

reference: https://github.com/ceph/ceph/blob/e879ce83c084f1a96e5dd0ab2b57bf909cfa423d/src/rgw/rgw_auth.cc#L461-L463

clang-tidy warning for `rgw_auth.cc`:
```
/home/suyash/ceph/src/rgw/rgw_auth.cc:493:10: warning: 'strategy_result' used after it was moved [bugprone-use-after-move]
  return strategy_result; 
         ^
/home/suyash/ceph/src/rgw/rgw_auth.cc:462:11: note: move occurred here
          strategy_handle_rejected(std::move(engine_result), policy,
          ^

```

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: src/rgw/driver/rados/rgw_user.cc
Comment (yuvalif) [inline src/rgw/driver/rados/rgw_user.cc:1558]: line 1554 seems like a real issue.
we move the `keys` parameter.
then we add new IDs into it at line 1558.
if the intent is to modify the key map into a new map, we should call `keys.clear()` after line 1554

**Comment by yuvalif** [comment]
@Suyashd999 please add the original warning/error from clag-tidy either into the commit comment (in case of false positives) or to the tracker issue if there is one.
or at least, pout it as a comment in the PR.

**Comment by Suyashd999** [comment]
> @Suyashd999 please add the original warning/error from clag-tidy either into the commit comment (in case of false positives) or to the tracker issue if there is one. or at least, pout it as a comment in the PR.

@yuvalif  I have added the original warning in the PR.

File: src/rgw/driver/rados/rgw_user.cc
Comment (Suyashd999) [reply src/rgw/driver/rados/rgw_user.cc:1558]: Hey @cbodley This issue was discussed in RGW standup can you please re-review this and provide an explanation on why this was marked as false-positive? Thank you.

**Comment by ronen-fr** [comment]
@Suyashd999 - please modify the PR description to match what the updated code does

File: src/rgw/driver/rados/rgw_user.cc
Comment (Suyashd999) [reply src/rgw/driver/rados/rgw_user.cc:1558]: @yuvalif It definitely is an issue. Also, adding `clear` after moving `keys`, clang-tidy no longer generates the warning.

**Comment by Suyashd999** [comment]
> must either fix the real problem some of us see in the code, then get Casey to agree with the analysis, or - explain why indeed there's no real bug here.
@ronen-fr 
It definitely is an issue. I have create a tracker and a PR for this:
tracker: https://tracker.ceph.com/issues/66553
PR:  https://github.com/ceph/ceph/pull/58119

File: src/rgw/driver/rados/rgw_user.cc
Comment (ronen-fr) [reply src/rgw/driver/rados/rgw_user.cc]: Note to the reader: the issue discussed in the comments above was removed from this PR.

**Review by ronen-fr — APPROVED** [review]

**Comment by ronen-fr** [comment]
LGTM. Requires RGW approval.
@ceph/rgw ?

**Comment by Suyashd999** [comment]
jenkins test api

**Review by yuvalif — APPROVED** [review]

**Comment by yuvalif** [comment]
jenkins test api

**Comment by ronen-fr** [comment]
@Suyashd999 , @yuvalif : I do not think a QA run is needed for an added comment. This can be merged once the CI test passes.

**Comment by yuvalif** [comment]
jenkins test api

**Comment by Suyashd999** [comment]
jenkins test api

**Comment by yuvalif** [comment]
jenkins test api

**Comment by yuvalif** [comment]
jenkins test api

**Comment by Suyashd999** [comment]
Hi @yuvalif!! Can we merge this?

**Pull request merged**
